File: Python/74. Search a 2D Matrix.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        m = len(matrix)
        n = len(matrix[0])
        l = 0
        r = m*n-1
        while l<=r:
            mid = int((l+r)/2)
            i=int(mid/n)
            j=int(mid%n)
            if matrix[i][j]==target:
                return True
            
            elif matrix [i][j]<target:
                l=mid + 1
            
            else:
                r=mid-1
            
    
        return False

File: C#/74. Search a 2D Matrix.cs
public class Solution {
    public bool SearchMatrix(int[][] matrix, int target) {
        int m = matrix.Length;
        int n = matrix[0].Length;
        int l = 0;
        int r = m*n-1;
        while (l<=r){
            int mid = (l+r)/2;
            int i=mid/n;
            int j=mid%n; 
            if (matrix[i][j]==target){
                return true;
            }
            else if(matrix [i][j]<target){
                l=mid + 1;
            }
            else{
                r=mid-1;
            }
        }
        return false;
    }
}
